src/main/sensors/gyro.c

```diff
@@ -580,6 +589,23 @@ STATIC_UNIT_TESTED void performGyroCalibration(gyroSensor_t *gyroSensor, uint8_t
 
 }
 
+uint32_t gyroSlewLimiter(int axis, gyroSensor_t *gyroSensor)
+{
+	// Probably should use gyroConfig()->gyro_slewlimit instead of 1<<14.
+	// Gate clip at 20 M/S gives rotation of say 45 deg at a 0.1M radius
+	// or a contact time of 5mS. Average yaw rate 9000 deg/S so gyro max
+	// rate probably exceeded almost instantly. So choose limit for
+	// impossible rate of change of rate.
+
+	int32_t newRawGyro = (int32_t) gyroSensor->gyroDev.gyroADCRaw[axis];
```

Comment:
Change to:
````
const int32_t newRawGyro = (int32_t) gyroSensor->gyroDev.gyroADCRaw[axis];
if (newRawGyro - gyroSensor->gyroDev.gyroADCRawP[axis] > (1<<14)) {
    newRawGyro = gyroSensor->gyroDev.gyroADCRawP[axis];
} else if (newRawGyro - gyroSensor->gyroDev.gyroADCRawP[axis] < (1<<14)) {
    newRawGyro = gyroSensor->gyroDev.gyroADCRawP[axis];
} else {
    gyroSensor->gyroDev.gyroADCRawP[axis] = newRawGyro;
}
To improve code efficiency.

Reply:
@martinbudden: This is on edge of over-optimization, at cost of readability  ...

Reply:
@ledvinap you're right. Deleting comment.